src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Tests/DependencyInjectionEventSourceTests.cs

```diff
@@ -6,31 +6,17 @@
 using System.Collections.Generic;
 using System.Diagnostics.Tracing;
 using System.Linq;
-using System.Reflection;
 using Microsoft.Extensions.DependencyInjection.Specification.Fakes;
 using Newtonsoft.Json.Linq;
 using Xunit;
 
 namespace Microsoft.Extensions.DependencyInjection.Tests
 {
-    [CollectionDefinition(nameof(EventSourceTests), DisableParallelization = true)]
-    public class EventSourceTests : ICollectionFixture<EventSourceTests>
+    public class DependencyInjectionEventSourceTests(
+        DependencyInjectionEventSourceTests.TestEventListenerFixture fixture)
+        : IClassFixture<DependencyInjectionEventSourceTests.TestEventListenerFixture>
     {
-    }
-
-    [Collection(nameof(EventSourceTests))]
-    public class DependencyInjectionEventSourceTests : IDisposable
-    {
-        private readonly TestEventListener _listener = new TestEventListener();
-
-        public DependencyInjectionEventSourceTests()
-        {
-            // clear the provider list in between tests
-            typeof(DependencyInjectionEventSource).GetField("_providers", BindingFlags.NonPublic | BindingFlags.Instance)
-                .SetValue(DependencyInjectionEventSource.Log, new List<WeakReference<ServiceProvider>>());
```

Comment:
Do you know what was the causing the random failure? Related to using WeakReference or Dispose?

Were you able to reproduce the test failures?

Reply:
The issue is that the test assumes it is the only one writing to the event source. This is not true because xunit executes the test collections/classes in parallel.

I was able to reproduce the error by inserting `Thread.Sleep` between the test clearing the event source and asserting on the events.